bitwarden_access_manager.py

```diff
@@ -0,0 +1,672 @@
+                )
+            else:
+                result = result + "group_access = []"
+        return result
+
+class BitwardenClient(NamedTuple):
+    connection: HTTPSConnection
+    bearer_token: str
+
+    @staticmethod
+    def new(client_id: str, client_secret: str) -> BitwardenClient:
+        response = requests.post(
+            "https://identity.bitwarden.com/connect/token",
+            headers={ "Content-Type": "application/x-www-form-urlencoded" },
+            data={
+                "grant_type": "client_credentials",
+                "scope": "api.organization",
+                "Accept": "application/json",
+            },
+            auth=(client_id, client_secret),
+        )
+        bearer_token = response.json()["access_token"]
+        return BitwardenClient(HTTPSConnection("api.bitwarden.com"), bearer_token)
+
+    def _http_get(self, url: str) -> HTTPResponse:
+        self.connection.request(
+            method="GET",
+            url=url,
+            headers={
+                "Accept": "application/json",
+                "Authorization": f"Bearer {self.bearer_token}",
+            },
+        )
+
+        return self.connection.getresponse()
+
+    def get_group(self, id: str) -> Any:
+        return json.load(self._http_get(f"/public/groups/{id}"))
+
+    def get_groups(self) -> Iterable[Group]:
+        data = self._http_get(f"/public/groups")
+        groups = json.load(data)
+        for group in groups["data"]:
+            yield Group(
+                id=group["id"],
+                name=group["name"],
+                access_all=group["accessAll"],
+            )
+
+    def get_collection_members(self, groups: Any) -> Iterable[MemberCollectionAccess]:
+            for group in groups:
+                group_id = group["id"]
+
+                data = self._http_get(f"/public/groups/{group_id}/member-ids")
+                member_ids = json.load(data)
+
+                for member_id in member_ids:
+                    data = self._http_get(f"/public/members/{member_id}")
+                    member = json.load(data)
+                    yield MemberCollectionAccess(
+                        name=member["name"],
+                    )
+
+    def get_collection_groups(self, groups: Any) -> Iterable[GroupCollectionAccess]:
+
+            for group in groups:
+                if group["readOnly"] == True:
+                    access = "readonly"
+                else:
+                    access = "write"
+
+                yield GroupCollectionAccess(
+                    name=self.get_group(group["id"])["name"],
+                    access=access,
+                )
+
+    def get_member_collections(self, members: Tuple[MemberCollectionAccess, ...]):
+        print("JERE")
+        for member in members:
+            data = self._http_get(f"/public/members/{member.id}")
+            member = json.load(data)
+            print(member)
+
+    def get_collection(self, id: str) -> Any:
```

Comment:
We should parse the result into a `Collection` type instead of returning the raw dict. Or is this not used and can we delete it?

Reply:
This is just a wrapper which is used in `get_collections`. I also remove `get_group` which was doing the same and `get_member_collections` which is not used anymore.

[c23a41c8e](https://github.com/ChorusOne/github-access-manager/pull/10/commits/c23a41c8ebda0c47cfb9c7584224d287c8c8c9e1)